session/proc_windows.go

```diff
@@ -0,0 +1,58 @@
+// +build windows
+
+package session
+
+import (
+	"os"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+)
+
+var procAttrs = &windows.SysProcAttr{
+	CreationFlags: windows.CREATE_UNICODE_ENVIRONMENT | windows.CREATE_NEW_PROCESS_GROUP,
+}
+
+// Windows doesn't understand SIGINT, instead we attach a console
+// to the process and send CTRL+BREAK / CTRL+C
+// Code from https://github.com/mattn/goreman/blob/master/proc_windows.go
+func terminateProc(process *os.Process) error {
+	dll, err := windows.LoadDLL("kernel32.dll")
+	if err != nil {
+		return err
+	}
+	defer dll.Release()
+
+	pid := process.Pid
+
+	f, err := dll.FindProc("AttachConsole")
+	if err != nil {
+		return err
+	}
+	r1, _, err := f.Call(uintptr(pid))
+	if r1 == 0 && err != syscall.ERROR_ACCESS_DENIED {
+		return err
+	}
+
+	f, err = dll.FindProc("SetConsoleCtrlHandler")
+	if err != nil {
+		return err
+	}
+	r1, _, err = f.Call(0, 1)
+	if r1 == 0 {
+		return err
+	}
+	f, err = dll.FindProc("GenerateConsoleCtrlEvent")
+	if err != nil {
+		return err
+	}
+	r1, _, err = f.Call(windows.CTRL_BREAK_EVENT, uintptr(pid))
+	if r1 == 0 {
+		return err
+	}
+	r1, _, err = f.Call(windows.CTRL_C_EVENT, uintptr(pid))
```

Comment:
Why send both `CTRL_BREAK_EVENT` and `CTRL_C_EVENT`? Are there cases where both are needed? I'm asking because it might be bad if the process receives a second interrupt while doing the cleanup caused by the first.

Reply:
Good point - the MSDN docs are somewhat confusing here, but I believe CTRL BREAK should be enough